dev/com.ibm.ws.microprofile.reactive.streams.operators_fat/test-applications/ConcurrentWorkTest/src/com/ibm/ws/microprofile/reactive/streams/test/concurrent/ReactiveParallelClientTest.java

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.reactive.streams.test.concurrent;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+
+import org.eclipse.microprofile.reactive.streams.operators.ProcessorBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+
+import componenttest.topology.utils.FATServletClient;
+
+@Path("/parallel")
+public class ReactiveParallelClientTest {
+
+    private static final int RS_ELEMENTS_IN_STREAM = 5;
+    private static final int DATA_ITEMS_SENT_DOWN_STREAM = 10;
+    private static int instanceId = 0;
+
+    /**
+     * This test is a simple reactive stream simulation that does some simple processing
+     * in a reactive stream before returning an answer.
+     *
+     * @param message
+     * @return
+     */
+    @GET
+    public String restEndpoint(@QueryParam("message") String message) {
+        System.out.println(message + " received on thread " + Thread.currentThread().getName());
+        singleStreamWorkload(RS_ELEMENTS_IN_STREAM, DATA_ITEMS_SENT_DOWN_STREAM);
+        return FATServletClient.SUCCESS;
+    }
+
+    /**
+     * This method can be used to simulate a small stream run
+     *
+     * @param rsProcessorsInStream    the length of the pipeline is altered by adding this many Processors
+     * @param dataItemsSentDownStream once the stream is running this many onData's will be sent down it before it completes
+     */
+    public <T> void singleStreamWorkload(int rsProcessorsInStream, int dataItemsSentDownStream) {
+
+        // start of stream
+        PublisherBuilder<Integer> streamHead = ReactiveStreams.fromIterable(() -> {
```

Comment:
The operation of these streams is complicated so might benefit from reworking with a few more comments.